### PR TITLE
Fixes 3846 Source Tabs and Gutter Column Don't Line Up

### PR DESCRIPTION
**Pull request by jainsneha23**
Associated Issue: #3846 

### Summary of Changes

* Pane toggle button padding changed to make it wider
* Margin left removed from first tab

### Test Plan

- [x] Open the debugger and check the view

### Screenshots/Videos (OPTIONAL)
screen shot 2017-10-18 at 11 12 15 pm

### CODE REVIEW

**Comment by jasonLaster** [comment]
nice... can you try and support cases where there are > 100 or > 1000 lines?

I think i have a comment in the issue for it too

**Comment by jainsneha23** [comment]
Sure..

**Comment by codehag** [comment]
This will probably have to be calculated based off of the codemirror width, do you need any help with that?

**Comment by jasonLaster** [comment]
here's where we calculate the gutter width now:

https://github.com/devtools-html/devtools-core/blob/master/packages/devtools-source-editor/src/utils.js#L40-L52

we could use it like this 

```js
import { SourceEditorUtils  } from "devtools-source-editor";
const { resizeBreakpointGutter } = SourceEditorUtils
```

**Comment by jasonLaster** [comment]
After seeing the core PR let's just stop using it :) and add getGW to the debugger utils

File: src/utils/editor/source-documents.js [hunk 1]
```diff
@@ -105,6 +107,23 @@ function showSourceText(editor: Object, source: Source) {
...
+  }
+}
+
+function resizeToggleButton(editor: Object) {
+  const toggleButton = document.querySelector(
+    ".source-header .toggle-button-start"
+  );
+  if (toggleButton) {
+    toggleButton.style.width = `${getLineNumberWidth(editor)}px`;
+  }
```
Comment (jasonLaster) [inline src/utils/editor/source-documents.js]: I think this is he Ron module for these functions. This m is or code mirror documents. Let's move them to utils/editor/index

Comment (jainsneha23) [reply src/utils/editor/source-documents.js]: sure

Comment (jainsneha23) [reply src/utils/editor/source-documents.js]: @jasonLaster I will move the code to debugger utils. Will you take care of cleaning up the function in core modules? I will close my PR then?

File: src/utils/ui.js [hunk 2]
```diff
@@ -3,7 +3,44 @@
...
  */
-export function isVisible() {
+function isVisible() {
   const el = document.querySelector("#mount");
   return el && el.getBoundingClientRect().width;
 }
+
+/* Gets the line numbers width in the code editor
+ */
+function getLineNumberWidth(editor) {
```
Comment (jasonLaster) [inline src/utils/ui.js]: small nit: lets add `export` to each function.

**Comment by jainsneha23** [comment]
screen shot 2017-10-22 at 11 37 38 am
screen shot 2017-10-22 at 11 37 53 am
screen shot 2017-10-22 at 11 38 06 am
screen shot 2017-10-22 at 11 37 08 am
screen shot 2017-10-22 at 11 36 56 am

**Comment by codehag** [comment]
Brilliant stuff!

**Pull request merged**